dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataverse/template/ManageTemplatesPage.java

```diff
@@ -0,0 +1,247 @@
+        }
+
+        return StringUtils.EMPTY;
+    }
+
+
+    public void makeDefault(Template templateIn) {
+        dataverse.setDefaultTemplate(templateIn);
+
+        templateService.updateDataverse(dataverse)
+                .onFailure(throwable -> JH.addMessage(FacesMessage.SEVERITY_FATAL, BundleUtil.getStringFromBundle("template.makeDefault.error")))
+                .onSuccess(dataverse -> JsfHelper.addFlashMessage(BundleUtil.getStringFromBundle("template.makeDefault")));
+    }
+
+    public void unselectDefault() {
+        dataverse.setDefaultTemplate(null);
+
+        templateService.updateDataverse(dataverse)
+                .onFailure(throwable -> JH.addMessage(FacesMessage.SEVERITY_FATAL, BundleUtil.getStringFromBundle("template.update.error")))
+                .onSuccess(dataverse -> JsfHelper.addFlashMessage(BundleUtil.getStringFromBundle("template.unselectDefault")));
+    }
+
+    public String cloneTemplate(Template templateIn) {
+
+        Try<Template> cloneTemplate = templateService.cloneTemplate(templateIn, dataverse, currentTime)
+                .andThenTry(() -> templateService.updateDataverse(dataverse));
+
+        if (cloneTemplate.isFailure()) {
+            JH.addMessage(FacesMessage.SEVERITY_FATAL, BundleUtil.getStringFromBundle("template.clone.error"));
+            return StringUtils.EMPTY;
+        }
+
+        templatesForView.add(cloneTemplate.get());
+
+        JsfHelper.addFlashMessage(BundleUtil.getStringFromBundle("template.clone"));
+        return "/template.xhtml?id=" + cloneTemplate.get().getId() + "&ownerId=" + dataverse.getId() + "&faces-redirect=true";
+    }
+
+    public void deleteTemplate() {
+        templateService.deleteTemplate(dataverse, selectedTemplate)
+                .onFailure(throwable -> JH.addMessage(FacesMessage.SEVERITY_FATAL, BundleUtil.getStringFromBundle("template.delete.error")))
+                .onSuccess(dataverse -> {
+                    JsfHelper.addFlashMessage(BundleUtil.getStringFromBundle("template.delete"));
+                    templatesForView.remove(selectedTemplate);
+                });
+
+    }
+
+    public void viewSelectedTemplate(Template selectedTemplate) {
+        this.selectedTemplate = selectedTemplate;
+
+        List<DatasetField> dsfForView = datasetFieldsInitializer.prepareDatasetFieldsForView(selectedTemplate.getDatasetFields());
+        mdbForView = DatasetFieldUtil.groupByBlock(dsfForView);
+    }
+
+    /**
+     * Updates dataverse regarding which templates it can use, since you can inherit templates from parent.
+     */
+    public String updateTemplatesRoot(AjaxBehaviorEvent event) throws AbortProcessingException {
+
+        if (dataverse.getOwner() != null) {
+
+            updateDefaultTemplates();
+
+            templateService.updateDataverseTemplate(dataverse, isInheritTemplatesValue())
+                    .onFailure(throwable -> Logger.getLogger(ManageTemplatesPage.class.getName()).log(Level.SEVERE, null, throwable))
```

Comment:
You sometimes log in service and sometime here. Shouldn't we have consistent approach to this?